.github/workflows/schema.yml

```diff
@@ -0,0 +1,42 @@
+name: Schema
+
+on:
+  push:
+    branches: [main]
+    paths:
+      - .github/workflows/generate.yml
+      - Makefile
+      - schema.yml
+
+  pull_request:
+    branches: [main]
+    paths:
+      - .github/workflows/generate.yml
+      - Makefile
+      - schema.yml
+
+jobs:
+  generate:
+    name: Generate clients
+    runs-on: ubuntu-latest
+
+    concurrency:
+      group: ${{ github.workflow }}-${{ github.ref }}
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Generate Javascript client
+        run: make javascript-client
+
+      - name: Generate Python client
+        run: make python-client
+
+      - name: Generate PHP client
+        run: make php-client
+
+      - name: Commit changes
+        if: github.repository_owner == 'libretime' && github.event_name == 'push'
+        uses: stefanzweifel/git-auto-commit-action@v4
+        with:
+          commit_message: "chore: update clients"
```

Comment:
This should probably also be configured to use @libretime-bot as the committer

Reply:
But libretime-bot is not reponsible for the push, it is only responsible for the schema push to this repository, or am I missing something ?

Reply:
I think we should probably be consistent across all our repos and use the bot for automated tasks. Currently the action uses the gh actions bot, while the new changes here make that the generic runner user

Reply:
I would set it only for the task it has the responsibility, as we will never be able to use the libretime bot for all automated tasks. It requires us to edit all our workflow to add non default values, for a benefit I don't yet see.

libretime/libretime => libretime/client uses the libretime-bot service account key to deploy changes, I understand the change in that case. But that would be too much hassle to edit all the other workflows.

Reply:
We're already using non-default values. For example https://github.com/libretime/client/blob/main/.github/workflows/generate.yml#L58 and https://github.com/libretime/libretime/blob/main/.github/workflows/api-schema.yml#L38

If you don't want to use the bot, then this at least needs updating to match the the main repo and the workflow this replaces

Reply:
Also that doesn't use the bot's token, we use a deploy key with write access

Reply:
https://github.com/libretime/libretime/blob/3d7083f60290812242b71bdde9dbd73882104c4f/.github/workflows/api-schema.yml#L38-L39

This was a work around waiting for https://github.com/stefanzweifel/git-auto-commit-action/pull/213

I kind of lost track which key is responsible for what, and you are probably right, that the libretime-bot account seem only used for the preview commands. And this somehow just show how confusing it will be if we start using the libretime-bot author were we don't even depend on any auth key from that bot.

Anyway, I just made the changes to use the libretime-bot to rebuild the client.